hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -681,8 +681,8 @@ protected void verifyBucketExists()
         trackDurationOfOperation(getDurationTrackerFactory(),
             STORE_EXISTS_PROBE.getSymbol(),
             () -> s3.doesBucketExist(bucket)))) {
-      throw new UnknownStoreException("Bucket " + bucket + " does not exist",
-          "/");
+      throw new UnknownStoreException("/", "Bucket " + bucket + " does not "
```

Comment:
sorry,I wasn't clear enough. Put in the bucket URI/name (with s3a:// ) as the path; the error message doesn't need to repeat it., e.g
```
new UnknownStoreException("s3a:// + bucket "/", "bucket does not exist"); same for below. 
```

Look at `PathIOE` to see how it constructs its exception...it adds the path before the message if non-null

Reply:
my bad didn't properly see what was being printed in this case.